crates/neon-macros/src/export/function/mod.rs

```diff
@@ -145,6 +148,28 @@ pub(super) fn export(meta: meta::Meta, input: syn::ItemFn) -> proc_macro::TokenS
     .into()
 }
 
+fn to_camel_case(name: &str) -> String {
```

Comment:
We could use the [regex](https://crates.io/crates/regex) crate for this. Granted, we're pretty conservative about taking dependencies, but it's a well-maintained crate, and it's just a compile-time dependency. ([example playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=670615e3baa025f239d6c4c77729f2f1))

```rust
impl Replacer for CamelReplacer {
    fn replace_append(&mut self, caps: &Captures<'_>, dst: &mut String) {
        dst.push_str(&caps[1].to_uppercase());
    }
}

let capture_the_snake = Regex::new(r"_(\p{Lowercase_Letter})").unwrap();
...
capture_the_snake.replace_all(name, CamelReplacer)
```

Even more straightforwardly, there's the [convert_case](https://crates.io/crates/convert_case) crate:

```rust
name.to_case(Case::Camel)
```

Reply:
Yeah I think it makes the most sense to use `convert_case`. It's a popular library, used by the AWS SDK and even by napi.rs for the same purpose as what we're doing here. And it'll handle edge cases (e.g. what if the identifier starts with an underscore) that we might not think of.

Reply:
I considered this and checked our existing tree to see if we already had any dependencies that did it. Since we did not, I opted not to use a dependency since it's only 15 lines of code.

You make a good point about edge cases. This implementation considers them and handles the properly. However, the definition of "proper" is a judgement call. For example, I chose to remove all underscores, but another library may choose to keep leading underscores or duplicates.

It might be worth using a library, not to save code, but to have confidence that we are consistent with the ecosystem. I do have one concern with using a library. This conversion is part of our **public** API and a change is a semver breaking change. Using a crate means we need to be careful when bumping dependency versions since a major version could change the conversion.

Do you think that semver risk is worth using a crate?